From 1d3ddab5897ecfd2f03675e76edadaa2db8d16aa Mon Sep 17 00:00:00 2001
From: Michael Steiner <michael.steiner@intel.com>
Date: Mon, 25 Mar 2019 20:54:41 -0700
Subject: [PATCH] sgx-enable invoked docker containers

---
 core/container/dockercontroller/dockercontroller.go | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/container/dockercontroller/dockercontroller.go b/core/container/dockercontroller/dockercontroller.go
index 2a709ed0b..51df8f7ff 100644
--- a/core/container/dockercontroller/dockercontroller.go
+++ b/core/container/dockercontroller/dockercontroller.go
@@ -168,6 +168,14 @@ func getDockerHostConfig() *docker.HostConfig {
 		CPUQuota:         getInt64("CpuQuota"),
 		CPUPeriod:        getInt64("CpuPeriod"),
 		BlkioWeight:      getInt64("BlkioWeight"),
+		Binds:            []string{"/var/run/aesmd:/var/run/aesmd"},
+		Devices: []docker.Device{
+			{
+				PathOnHost:        "/dev/isgx",
+				PathInContainer:   "/dev/isgx",
+				CgroupPermissions: "rwm",
+			},
+		},
 	}
 }
 
@@ -182,6 +190,7 @@ func (vm *DockerVM) createContainer(client dockerClient, imageID, containerID st
 			Env:          env,
 			AttachStdout: attachStdout,
 			AttachStderr: attachStdout,
+			Volumes:      map[string]struct{}{"/var/run/aesmd": {}},
 		},
 		HostConfig: getDockerHostConfig(),
 	})
-- 
2.17.1

