diff --git i/core/container/dockercontroller/dockercontroller.go w/core/container/dockercontroller/dockercontroller.go
index 2a709ed0b..51df8f7ff 100644
--- i/core/container/dockercontroller/dockercontroller.go
+++ w/core/container/dockercontroller/dockercontroller.go
@@ -168,6 +168,14 @@ func getDockerHostConfig() *docker.HostConfig {
 		CPUQuota:         getInt64("CpuQuota"),
 		CPUPeriod:        getInt64("CpuPeriod"),
 		BlkioWeight:      getInt64("BlkioWeight"),
+		Binds:            []string{"/var/run/aesmd:/var/run/aesmd"},
+		Devices: []docker.Device{
+			{
+				PathOnHost:        "/dev/isgx",
+				PathInContainer:   "/dev/isgx",
+				CgroupPermissions: "rwm",
+			},
+		},
 	}
 }
 
@@ -182,6 +190,7 @@ func (vm *DockerVM) createContainer(client dockerClient, imageID, containerID st
 			Env:          env,
 			AttachStdout: attachStdout,
 			AttachStderr: attachStdout,
+			Volumes:      map[string]struct{}{"/var/run/aesmd": {}},
 		},
 		HostConfig: getDockerHostConfig(),
 	})
