diff --git a/ubuntu-xenial.dockerfile b/ubuntu-xenial.dockerfile
index dca51eb..697d4cb 100644
--- a/ubuntu-xenial.dockerfile
+++ b/ubuntu-xenial.dockerfile
@@ -1,9 +1,9 @@
-FROM tozd/runit:ubuntu-xenial
+FROM hyperledger/fabric-ccenv:latest
 
 COPY ./patches /patches
 
 RUN apt-get update -q -q && \
- apt-get install wget python git patch build-essential ocaml automake autoconf libtool libssl-dev libcurl4-openssl-dev protobuf-compiler protobuf-c-compiler libprotobuf-dev libprotobuf-c0-dev alien uuid-dev libxml2-dev cmake pkg-config --yes --force-yes && \
+ apt-get install wget python git patch build-essential ocaml automake autoconf libtool libssl-dev libcurl4-openssl-dev protobuf-compiler protobuf-c-compiler libprotobuf-dev libprotobuf-c0-dev alien uuid-dev libxml2-dev cmake pkg-config libsystemd-dev --yes --force-yes && \
  mkdir -p /tmp/icls && \
  cd /tmp/icls && \
  wget http://registrationcenter-download.intel.com/akdlm/irc_nas/11414/iclsClient-1.45.449.12-1.x86_64.rpm && \
@@ -18,7 +18,7 @@ RUN apt-get update -q -q && \
  make install && \
  rm -rf /tmp/dynamic-application-loader-host-interface && \
  cd /tmp && \
- git clone https://github.com/01org/linux-sgx.git && \
+ git clone -b sgx_2.4 https://github.com/01org/linux-sgx.git && \
  cd / && \
  for patch in /patches/*; do patch --prefix=/patches/ -p0 --force "--input=$patch" || exit 1; done && \
  rm -rf /patches && \
